budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -1212,3 +1224,248 @@ def test_correct_project_returned(self):
         self.assertEqual(len(results), 1)
         self.assertEqual(results[0]["name"], "Blue School")
         self.assertNotContains(response, "Red School")
+        self.assertEqual(
+            response.data["csv_download_url"],
+            "/infrastructure-projects/provincial/search/csv?q=Eastern+Cape+School",
+        )
+
+
+class ProvInfraProjectIRMSnapshotCSVDownloadMixin:
+    def _test_csv_content_correctness(self, csv_reader, items_to_compare):
+        for index, row in enumerate(csv_reader):
+            self.assertListEqual(
+                list(row.keys()), ProvInfraProjectCSVSerializer.Meta.fields
+            )
```

Comment:
```suggestion
            # Verify all the serializer fields are present in the CSV
            self.assertListEqual(
                list(row.keys()), ProvInfraProjectCSVSerializer.Meta.fields
            )
```

Reply:
Please add comments for what the rest of the assertions verify since the abstraction makes it hard to know exactly what the test is checking.

I find that number 1 priority for tests is to easily see what is being tested (and what isn't) without executing code in your head